compiler/src/iree/compiler/Codegen/Common/test/decompose_linalg_generic.mlir

```diff
@@ -1,81 +1,21 @@
-      iterator_types = ["parallel", "parallel"]}
-      ins(%arg0 : tensor<64xf32>) outs(%arg1 : tensor<?x64xf32>) {
-  ^bb0(%arg2: f32, %arg3: f32):
-    linalg.yield %arg3 : f32
-  } -> tensor<?x64xf32>
-  func.return %12 : tensor<?x64xf32>
-}
-
 // -----
-// CHECK: #map0 = affine_map<(d0, d1) -> (d1)>
-// CHECK: #map1 = affine_map<(d0, d1) -> (d0, d1)>
 // CHECK-LABEL: @parallel_with_broadcast_dynamic
 func.func @parallel_with_broadcast_dynamic(
     %arg0 : tensor<64xf32>, %arg1 : tensor<?x64xf32>) -> tensor<?x64xf32> {
-  //      CHECK: %[[C0:.*]] = arith.constant 0 : index
-  //      CHECK: %[[D0:.*]] = tensor.dim %arg1, %[[C0]] : tensor<?x64xf32>
-  //
-  // Generic0: Decomposed addf
-  //      CHECK: %[[IT0:.*]] = linalg.init_tensor [%[[D0]], 64] : tensor<?x64xf32>
-  //      CHECK: %[[G0:.*]] = linalg.generic
-  // CHECK-SAME:   {indexing_maps = [#map0, #map1, #map1], iterator_types = ["parallel", "parallel"]}
-  // CHECK-SAME:   ins(%arg0, %arg1 : tensor<64xf32>, tensor<?x64xf32>) outs(%[[IT0]] : tensor<?x64xf32>)
-  //      CHECK:   ^bb0(%[[BARG0_0:.*]]: f32, %[[BARG_0_1:.*]]: f32, %[[BARG_0_2:.*]]: f32):
-  //      CHECK:     %[[V0:.*]] = arith.addf %[[BARG0_0]], %[[BARG_0_1]] : f32
-  //      CHECK:     linalg.yield %[[V0]]
-  //
-  // Generic1: Decomposed mulf
-  //      CHECK: %[[IT1:.*]] = linalg.init_tensor [%[[D0]], 64] : tensor<?x64xf32>
-  //      CHECK: %[[G1:.*]] = linalg.generic
-  // CHECK-SAME:   {indexing_maps = [#map1, #map0, #map1], iterator_types = ["parallel", "parallel"]}
-  // CHECK-SAME:   ins(%[[G0]], %arg0 : tensor<?x64xf32>, tensor<64xf32>) outs(%[[IT1]] : tensor<?x64xf32>)
-  //      CHECK:   ^bb0(%[[BARG1_0:.*]]: f32, %[[BARG_1_1:.*]]: f32, %[[BARG_1_2:.*]]: f32):
-  //      CHECK:     %[[V1:.*]] = arith.mulf %[[BARG1_0]], %[[BARG_1_1]] : f32
-  //      CHECK:     linalg.yield %[[V1]]
-  //
-  // Degenerate final copy.
-  // Note that canonicalization will collapse this with the previous
-  // in this case but emitting keeps us fully general.
-  //      CHECK: %[[G2:.*]] = linalg.generic
-  // CHECK-SAME:   {indexing_maps = [#map1, #map1], iterator_types = ["parallel", "parallel"]}
-  // CHECK-SAME:   ins(%[[G1]] : tensor<?x64xf32>) outs(%arg1 : tensor<?x64xf32>)
-  //      CHECK:   ^bb0(%[[BARG2_0:.*]]: f32, %[[BARG2_1:.*]]: f32):
-  //      CHECK:     linalg.yield %[[BARG2_0]]
-  //      CHECK: return %[[G2]]
+  // Just verify that the upstream patterns are working. They should produce
+  // two generics: one with an add and the next with a mul
+  // CHECK: linalg.generic
+  // CHECK: arith.addf
+  // CHECK: linalg.generic
+  // CHECK: arith.mulf
   %12 = linalg.generic {
       indexing_maps = [affine_map<(d0, d1) -> (d1)>, affine_map<(d0, d1) -> (d0, d1)>],
       iterator_types = ["parallel", "parallel"]}
       ins(%arg0 : tensor<64xf32>) outs(%arg1 : tensor<?x64xf32>) {
   ^bb0(%arg2: f32, %arg3: f32):
-    %13 = arith.addf %arg2, %arg3 : f32
```

Comment:
The semantics of reading the argument for the outs when everything is parallel is not clear to me (if you are read-modify-write-ing you should have a "reduction" iterator type). Thats why the upstream probably doesnt handle this. If needed I can fix that.

Reply:
I get something like this with "bias adds":

```
    linalg.fill ins(%cst : f32) outs(%3 : memref<2x1xf32>)
    linalg.matmul {lowering_config = #iree_codegen.lowering_config<tile_sizes = [[2, 0, 0]]>} ins(%0, %1 : memref<2x3xf32>, memref<3x1xf32>) outs(%3 : memref<2x1xf32>)
    linalg.generic {indexing_maps = [affine_map<(d0, d1) -> ()>, affine_map<(d0, d1) -> (d0, d1)>], iterator_types = ["parallel", "parallel"]} ins(%2 : memref<f32>) outs(%3 : memref<2x1xf32>) {                                ^bb0(%arg1: f32, %arg2: f32):                                                                                                                                                                                                  %6 = arith.addf %arg1, %arg2 : f32                                                                                                                                                                                           linalg.yield %6 : f32
    }
```

But I don't have to deal with them right now. And there may be something not right at the source.

Reply:
(these come from a tosa fully_connected)

Reply:
ok, I see why this is happening. This is an artifact of having destination passing style and getting the computation bufferized in place without allocating scratch memory. I'll fix this issue upstream.